src/main/java/com/icthh/xm/uaa/service/dto/OtpSendDTO.java

```diff
@@ -0,0 +1,39 @@
+package com.icthh.xm.uaa.service.dto;
+
+import com.icthh.xm.uaa.domain.properties.TenantProperties;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
+
+@Builder
+@Getter
+@Setter
+@AllArgsConstructor
+public class OtpSendDTO {
+
+    private String otp;
+    private String destination;
+    private String userKey;
+    private TenantProperties.NotificationChannel channel;
+
+    public OtpSendDTO(String otp, String destination, String userKey) {
+        this.otp = otp;
+        this.destination = destination;
+        this.userKey = userKey;
+    }
+
+    public void setTitleKey(String titleKey) {
+        if (channel == null) {
```

Comment:
it`s acceptable here to extract null handle to getter, and use getChannel().setTitleKey

Reply:
this was added to support previous tfa otp logic only

Reply:
since it a long term existing dto, i guess that it shouldn't harm

Reply:
made an update to avoid questioning this